core/trino-main/src/main/java/io/trino/operator/DirectExchangeClient.java

```diff
@@ -84,6 +89,11 @@
     @GuardedBy("this")
     private boolean closed;
 
+    @GuardedBy("this")
+    private final TDigest clientWaitingTimeInMillis = new TDigest();
```

Comment:
Could you add that metric to `HttpPageBufferClient` (and `PageBufferClientStatus`) because we can track it on lower level.
Please merge individual metrics in `DirectExchangeClientStatus`

This should be called `responseTimeInMillis`

Reply:
> This should be called responseTimeInMillis

I do not think so. We do not measure `responseTime` here - it should be resposibility of http client or lower layers, not this one. 
It measures the distribution of client's waiting time to be pulled from the queue. 

> Could you add that metric to HttpPageBufferClient (and PageBufferClientStatus) because we can track it on lower level.

This metric is specific to the `DirectExchangeClient` so I do not think that we can add that metric to other places.

Reply:
> This metric is specific to the DirectExchangeClient so I do not think that we can add that metric to other places.

It's really `HttpPageBufferClient` that performs requests, and `responseTime` can be measured for on that request. `HttpPageBufferClient` is really the lowest place where it can be measured (and not `DirectExchangeClient`)